### ISSUE TITLE
Calculating $(_TargetFrameworkVersionWithoutV) costs 0.7% of a evaluation of solution-wide design-time build

### ISSUE THREAD

**Issue by davkean** [dotnet/sdk#1587]
See: https://github.com/dotnet/sdk/issues/1586

Pass|File|Line #|Expression|Inc (ms)|Inc (%)|Exc (ms)|Exc (%)|#|Bug
---|---|---:|---|---:|---:|---:|---:|---:|---
Total Evaluation||||6159|100%|54|0.9%||
Properties (Pass 1)||||3312|53.8%|9|0.1%||
ItemDefinitionGroup (Pass 2)||||14|0.2%|2|0%||
Items (Pass 3)||||566|9.2%|24|0.4%||
Lazy Items (Pass 3.1)||||1826|29.6%|158|2.6%||
UsingTasks (Pass 4)||||59|1%|59|1%||
Targets (Pass 5)||||327|5.3%|208|3.4%||
Properties (Pass 1)|Microsoft.NET.Build.Extensions.targets|18|`<_TargetFrameworkVersionWithoutV Condition="$(TargetFrameworkVersion.StartsWith('v'))" >$(TargetFram...`|45|0.7%|42|0.7%|223|

**Comment by davkean** [comment]
Note that this bumped up to 1.1% after removing the unneeded NETStandard.Library.NETFramework package.

**Comment by nguerrera** [comment]
Every way that I've tried to rewrite this has been slower than what we already have. I tried passing Ordinal to StartsWith and using an integer intrinsic function (BitwiseXor(1, 0) instead of literal 1 in the hopes it would help the binder see that there is no string arg. Both made things slower. I then tried checking Length > 0 and [0] == 'v' instead of StartsWith. Worse again.

It seems like just about any property function call is really expensive. I'm seeing 0.3 ms inclusive for 1 project on this one. We need to fix this in the binder.

I don't think we're going to get anywhere trying to optimize these individual calls.
